gemma-core/src/main/java/ubic/gemma/persistence/service/expression/designElement/CompositeSequenceServiceImpl.java

```diff
@@ -84,6 +85,33 @@ public CompositeSequenceServiceImpl( CompositeSequenceDao compositeSequenceDao,
         return this.compositeSequenceDao.findByGene( gene );
     }
 
+    /**
+     * Include gene mapping summary in the {@link CompositeSequenceValueObject}.
+     */
+    @Override
+    @Transactional(readOnly = true)
+    public CompositeSequenceValueObject loadValueObject( CompositeSequence cs ) {
+        CompositeSequenceValueObject vo = super.loadValueObject( cs );
+        // Not passing the vo since that would create data redundancy in the returned structure
+        vo.setGeneMappingSummaries(
+                this.getGeneMappingSummary( this.bioSequenceService.findByCompositeSequence( cs ), null ) );
+        return vo;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public CompositeSequenceValueObject loadValueObjectWithoutGeneMappingSummary( CompositeSequence cs ) {
+        return super.loadValueObject( cs );
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public List<CompositeSequenceValueObject> loadValueObjectsWithoutGeneMappingSummary( Collection<CompositeSequence> compositeSequences ) {
+        return compositeSequences.stream()
+                .map( this::loadValueObjectWithoutGeneMappingSummary )
+                .collect( Collectors.toList() );
+    }
+
```

Comment:
This is the alternative if you want to convert multiple models efficiently  by avoiding the transactional overhead.